lib/ain-grpc/src/transaction_request.rs

```diff
@@ -0,0 +1,99 @@
+use crate::bytes::Bytes;
+use ethereum::{
+    AccessListItem, EIP1559TransactionMessage, EIP2930TransactionMessage, LegacyTransactionMessage,
+};
+use ethereum_types::{H160, U256};
+use serde::{Deserialize, Serialize};
+
+pub enum TransactionMessage {
+    Legacy(LegacyTransactionMessage),
+    EIP2930(EIP2930TransactionMessage),
+    EIP1559(EIP1559TransactionMessage),
+}
+
+/// Transaction request coming from RPC
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionRequest {
```

Comment:
Can we re-use `CallRequest` or keep only this one instead ?

Reply:
hmm.. yea i have doubt on this earlier.. question.. why splitting 1 struct to 2 structs since they contain same field (slightly diff)..
btw i think current design is make more sense.. obviously splitting into 2 structs is to serve for own specific purpose.. readability and easier to maintain